src-self-hosted/cgen.zig

```diff
@@ -133,6 +147,28 @@ pub fn generate(file: *C, decl: *Decl) !void {
                                 return file.fail(decl.src(), "TODO non-constant call inst?", .{});
                             }
                         },
+                        .ret => {
```

Comment:
Can you start separating these prongs into their own functions? This function is getting a bit hard to review (hard to scroll up and see the function signature). See the example set by codegen.zig: 

https://github.com/ziglang/zig/blob/7adbc1140373548f2985c8054b540cc5648f6857/src-self-hosted/codegen.zig#L415-L435

Reply:
Yeah, definitely